### PR TITLE
feat: support stop --title

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- close https://github.com/eggjs/egg/issues/1735
- close https://github.com/eggjs/egg/issues/1733
- relate: https://github.com/eggjs/egg-scripts/pull/12#issuecomment-344158657
- remove `stop baseDir` support
- ~~change egg to 2~~

### CODE REVIEW

File: lib/cmd/stop.js [hunk 1]
```diff
@@ -32,12 +38,15 @@ class StopCommand extends Command {
     if (!path.isAbsolute(baseDir)) baseDir = path.join(context.cwd, baseDir);
     argv.baseDir = baseDir;
 
+    const pkgInfo = require(path.join(baseDir, 'package.json'));
+    argv.title = argv.title || `egg-server-${pkgInfo.name}`;
```
Comment (popomore) [inline lib/cmd/stop.js]: 不要默认吧，stop 应该全杀，不然改了 package name 就杀不掉进程了。

可以改成 title 不存在就不传。

Comment (popomore) [reply lib/cmd/stop.js]: 可以加个这个用例

**Comment by atian25** [comment]
改了，干掉 baseDir 的支持。

File: lib/cmd/stop.js [hunk 2]
```diff
@@ -25,19 +31,12 @@ class StopCommand extends Command {
...
-    argv.baseDir = baseDir;
-
-    this.logger.info(`stopping egg application at ${baseDir}`);
+    this.logger.info(`stopping egg application ${argv.title ? `with --title=${argv.title}` : ''}`);
 
     // node /Users/tz/Workspaces/eggjs/egg-scripts/lib/start-cluster {"title":"egg-server","workers":4,"port":7001,"baseDir":"/Users/tz/Workspaces/eggjs/test/showcase","framework":"/Users/tz/Workspaces/eggjs/test/showcase/node_modules/egg"}
     let processList = yield this.helper.findNodeProcess(item => {
       const cmd = item.cmd;
-      return cmd.includes('start-cluster');
+      return cmd.includes('start-cluster') && (!argv.title || cmd.includes(`"title":"${argv.title}"`));
```
Comment (popomore) [inline lib/cmd/stop.js]: 这个逻辑看着有点绕，还是先判断 title 吧

```js
argv.title ?
  cmd.includes('start-cluster') && cmd.includes(`"title":"${argv.title}"`) :
  cmd.includes('start-cluster')
```

Comment (atian25) [reply lib/cmd/stop.js]: 改了，下面的也一起改了

**Comment by popomore** [comment]
+ egg-scripts@2.3.0

**Pull request merged**